src/mlpack/core/data/image_resize_crop.hpp

```diff
@@ -0,0 +1,135 @@
+    const size_t newWidth, const size_t newHeight)
+{
+  // First check if we are resizing one image or a group of images, the check
+  // is going to be different depending on the dimension.
+  // If the user would like to resize a set of images of different dimensions,
+  // then they need to consider passing them image by image. Otherwise, as
+  // assume that all images have identical dimension and need to be resized.
+  if (images.n_cols == 1)
+  {
+    if (images.n_elem != (info.Width() * info.Height() * info.Channels()))
+    {
+      Log::Fatal << "ResizeImages(): dimensions mismatch: the number of pixels "
+          << "is not equal to the dimension provided by the given ImageInfo."
+          << std::endl;
+    }
+  }
+  else
+  {
+    if (images.n_rows != (info.Width() * info.Height() * info.Channels()))
+    {
+      Log::Fatal << "ResizeImages(): dimension mismatch: in the case of "
+          << "several images, please check that all the images have the same "
+          << "dimensions; if not, load each image in one column and call this "
+          << "function iteratively." << std::endl;
+    }
+  }
+
+  stbir_pixel_layout channels;
+  if (info.Channels() == 1)
+  {
+    channels = STBIR_1CHANNEL;
+  }
+  else if (info.Channels() == 3)
+  {
+    channels = STBIR_RGB;
+  }
+
+  // This is required since STB only accept unsigned chars.
+  // set the new matrix size for copy
+  size_t newDimension = newWidth * newHeight * info.Channels();
+  arma::Mat<float> resizedFloatImages;
+  arma::Mat<unsigned char> resizedImages;
+
+  // This is not optimal, but I do not want to allocate memory for nothing.
+  if (std::is_same<eT, float>::value)
+    resizedFloatImages.set_size(newDimension, images.n_cols);
+  else
+    resizedImages.set_size(newDimension, images.n_cols);
+
+  for (size_t i = 0; i < images.n_cols; ++i)
+  {
+    if constexpr (std::is_same<eT, unsigned char>::value)
+    {
+      arma::Mat<unsigned char> tempDest(newDimension, 1);
+      stbir_resize_uint8_linear(images.colptr(i), info.Width(), info.Height(),
+          0, tempDest.memptr(), newWidth, newHeight, 0, channels);
+
+      resizedImages.col(i) = std::move(tempDest);
+    }
+    else if constexpr (std::is_same<eT, float>::value)
+    {
+      arma::fmat tempDestFloat(newDimension, 1);
+      stbir_resize_float_linear(images.colptr(i), info.Width(), info.Height(),
+          0, tempDestFloat.memptr(), newWidth, newHeight, 0, channels);
+      resizedFloatImages.col(i) = std::move(tempDestFloat);
+    }
+    else
+    {
+      arma::Mat<unsigned char> tempSrc =
+          arma::conv_to<arma::Mat<unsigned char>>::from(images);
+
+      arma::Mat<unsigned char> tempDest(newDimension, 1);
+
+      stbir_resize_uint8_linear(tempSrc.colptr(i), info.Width(), info.Height(),
+          0, tempDest.memptr(), newWidth, newHeight, 0, channels);
+
+      resizedImages.col(i) = arma::conv_to<arma::Mat<eT>>::from(tempDest);
+    }
+  }
+
+  images = std::move(resizedImages);
```

Comment:
```suggestion
  images = arma::conv_to<Mat<eT>>::from(std::move(resizedImages));
```
We need to put the `conv_to` here, because `resizedImages` has type `arma::Mat<unsigned char>` and `images` has type `arma::Mat<eT>`.